components/layout/display_list_builder.rs

```diff
@@ -1190,6 +1192,30 @@ impl FragmentDisplayListBuilding for Fragment {
             transform = pre_transform.mul(&transform).mul(&post_transform);
         }
 
+        let mut perspective = Matrix4::identity();
+        match self.style().get_effects().perspective {
+            LengthOrNone::Length(d) => {
```

Comment:
Actually, come to think of it, couldn't this be `let perspective = match self.style().get_effects().perspective {` ... and avoid the mutability?